src/audio/mux/mux.c

```diff
@@ -140,15 +140,118 @@ static void mux_free(struct comp_dev *dev)
 	rfree(dev);
 }
 
+static uint8_t get_stream_index(struct comp_data *cd, uint32_t pipe_id)
+{
+	int i;
+
+	for (i = 0; i < MUX_MAX_STREAMS; i++)
+		if (cd->config.streams[i].pipeline_id == pipe_id)
+			return i;
+
+	trace_mux_error("get_stream_index() error: couldn't find configuration "
+			"for connected pipeline %u", pipe_id);
+
+	return 0;
+}
+
+static int mux_verify_params(struct comp_dev *dev,
+			     struct sof_ipc_stream_params *params)
+{
+	struct list_item *clist;
+	struct comp_buffer *sinkb;
+	struct comp_buffer *sourceb;
+
+	tracev_mux_with_ids(dev, "mux_verify_params()");
+
+	sinkb = list_first_item(&dev->bsink_list, struct comp_buffer,
+				source_list);
+
+	if (dev->direction == SOF_IPC_STREAM_PLAYBACK) {
+		params->channels = sinkb->channels;
+
+		/* set sink buffer params */
+		buffer_set_params(sinkb, params);
+
+	} else {
+		list_for_item(clist, &dev->bsource_list) {
+			sourceb = container_of(clist, struct comp_buffer,
+					       sink_list);
+
+			params->channels = sourceb->channels;
+
+			/* set source buffer params */
+			buffer_set_params(sinkb, params);
+		}
+	}
+
+	/* set component period frames */
+	component_set_period_frames(dev, sinkb->rate);
```

Comment:
I'm seeing a lot of duplication in _verify_params() depending on count of source/sink buffers and direction. Can we consolidate this to a public helper API.

Reply:
I will work on that